internal/starlark/db.go

```diff
@@ -2,251 +2,17 @@ package starlark
-		ew := errWriter{}
-		m := ew.assertModel(mn, tx)
-		ma := ew.assertMatcher(mm)
-		if ew.err != nil {
-			return nil, ew.err
-		}
-		t := tx.Ref(m.ID())
-		results := tx.Query(t, db.Filter(t, ma)).All()
-		var out []Record
-		for _, res := range results {
-			out = append(out, &starlarkRecord{inner: res.Record})
-		}
-		return out, nil
-	}
-	env["Eq"] = func(k, v interface{}) (db.Matcher, error) {
-		ew := errWriter{}
-		key := ew.assertString(k)
-		if ew.err != nil {
-			return nil, ew.err
-		}
-		return db.Eq(key, v), nil
-	}
-	env["And"] = func(matchers ...interface{}) (db.Matcher, error) {
-		ew := errWriter{}
-		var out []db.Matcher
-		for i := 0; i < len(matchers); i++ {
-			m := ew.assertMatcher(matchers[i])
-			if ew.err != nil {
-				return nil, ew.err
-			}
-			out = append(out, m)
-		}
-		return db.And(out...), nil
-	}
-	env["Insert"] = func(mn interface{}, fields interface{}) (Record, error) {
-		ew := errWriter{}
-		m := ew.assertModel(mn, tx)
-		r, err := tx.MakeRecord(m.ID())
-		if err != nil {
-			return nil, err
-		}
-		ew.SetDBRecord("id", uuid.New(), r)
-		fieldMap := ew.assertMap(fields)
-		for key, val := range fieldMap {
-			ks := ew.assertString(key)
-			ew.SetDBRecord(ks, val, r)
-		}
-		if ew.err != nil {
-			return nil, ew.err
-		}
-		tx.Insert(r)
-		return &starlarkRecord{inner: r}, nil
-	}
-	env["Update"] = func(r interface{}, fields interface{}) (Record, error) {
-		ew := errWriter{}
-		rec := ew.assertStarlarkRecord(r)
-		if ew.err != nil {
-			return nil, ew.err
-		}
-		oldRec := rec.inner
-		newRec := oldRec.DeepCopy()
-		fieldMap := ew.assertMap(fields)
-		for key, val := range fieldMap {
-			ks := ew.assertString(key)
-			ew.SetDBRecord(ks, val, newRec)
-		}
-		if ew.err != nil {
-			return nil, ew.err
-		}
-		err := tx.Update(oldRec, newRec)
-		if err != nil {
-			return nil, err
-		}
-		return &starlarkRecord{inner: newRec}, err
-
-	}
-	env["Connect"] = func(s interface{}, r1 interface{}, r2 interface{}) (bool, error) {
-		ew := errWriter{}
-		bname := ew.assertString(s)
-		rec1 := ew.assertStarlarkRecord(r1)
-		rec2 := ew.assertStarlarkRecord(r2)
-		if ew.err != nil {
-			return false, ew.err
-		}
-
-		var relationship db.Relationship
-		rels, err := rec1.inner.Interface().Relationships()
-		if err != nil {
-			return false, err
-		}
-		found := false
-		for _, rel := range rels {
-			if rel.Name() == bname {
-				relationship = rel
-				found = true
-				break
-			}
-		}
-		if !found {
-			return false, err
-		}
-
-		err = tx.Connect(rec1.inner.ID(), rec2.inner.ID(), relationship.ID())
-		if err != nil {
-			return false, err
-		}
-		return true, nil
-	}
-	env["Delete"] = func(r interface{}) (Record, error) {
-		ew := errWriter{}
-		rec := ew.assertStarlarkRecord(r)
-		if ew.err != nil {
-			return nil, ew.err
-		}
-		err := tx.Delete(rec.inner)
-		if err != nil {
-			return nil, err
-		}
-		return rec, err
-	}
-	env["Parse"] = func(code interface{}) (string, bool, error) {
+	env["aft"] = handlers.Aft(tx)
```

Comment:
Rename this to "api", I think -- the api is just one part of aft!

Reply:
yep~